pandas/tests/arrays/string_/test_string.py

```diff
@@ -345,3 +345,22 @@ def test_astype_from_float_dtype(dtype):
     result = s.astype("string")
     expected = pd.Series(["0.1"], dtype="string")
     tm.assert_series_equal(result, expected)
+
+
+def test_contains():
+    # GH-37867
+    arr = pd.arrays.StringArray(np.array(["a", "b"], dtype=object))
+
+    assert "a" in arr
+    assert "x" not in arr
+    assert pd.NA not in arr
+
+    arr = pd.arrays.StringArray(np.array(["a", pd.NA]))
+    assert "a" in arr
+    assert "x" not in arr
```

Comment:
np.nan should be True here?

Reply:
or False? 
Should we distinguish between np.nan vs pd.NA ? (currently the indexing engines do distinguish both)

Reply:
They all (np.nan, pd.NA and pd.NaT) True here. The reason is that:

```python
>>> pd.core.dtypes.missing.is_valid_nat_for_dtype(pd.NaT, pd.StringDtype())
True
```

This looks like a different bug than this PR.

Reply:
It's a direct consequence of the implementation, and I think actually easy to fix. Eg maybe you can use `if val is self.dtype.na_value` instead of `isna(val)` or `is_valid_nat_for_dtype(val, ..)` ?